code/modules/projectiles/guns/special/blastcannon.dm

```diff
@@ -304,37 +300,51 @@
 	var/medium_ex_range = 0
 	/// The maximum distance this will inflict [EXPLODE_LIGHT]
 	var/light_ex_range = 0
+	/// Whether or not to care about the explosion block of the things we are passing through.
+	var/reactionary
 
-/obj/projectile/blastwave/Initialize(mapload, heavy_ex_range, medium_ex_range, light_ex_range)
+/obj/projectile/blastwave/Initialize(mapload, heavy_ex_range, medium_ex_range, light_ex_range, reactionary = CONFIG_GET(flag/reactionary_explosions))
 	range = max(heavy_ex_range, medium_ex_range, light_ex_range, 0)
 	src.heavy_ex_range = heavy_ex_range
 	src.medium_ex_range = medium_ex_range
 	src.light_ex_range = light_ex_range
+	src.reactionary = reactionary
 	return ..()
 
 /obj/projectile/blastwave/Range()
 	. = ..()
 	if(QDELETED(src))
 		return
 
-	heavy_ex_range = max(heavy_ex_range - 1, 0)
-	medium_ex_range = max(medium_ex_range - 1, 0)
-	light_ex_range = max(light_ex_range - 1, 0)
-
-	if(heavy_ex_range)
-		SSexplosions.highturf += loc
+	var/decrement = 1
+	var/atom/location = loc
+	if (reactionary)
+		var/atom/location = loc
```

Comment:
double definition

Reply:
Yeah. I noticed. I'll fix it when I'm done testing a fix for the bug mentioned in #63859.

Reply:
Done.